### PR TITLE
Chore/mui

### PR DESCRIPTION
**Pull request by fortunato**
- upgrade form material-ui to mui
- upgrade from pegged jsonforms alpha release to latest beta (mui dependency)
- npm update
- webpack v4 to v5 upgrade
- updated babel packages
- updated storybook packages
- updated .nvmrc to latest LTS

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,33 +1,34 @@
...
+      jsx: true // Allows for the parsing of JSX
+
+    }
+  },
+  parser: '@typescript-eslint/parser',
+  plugins: ['react', '@typescript-eslint', 'testing-library'],
+  rules: {
+    '@typescript-eslint/ban-ts-ignore': 'off',
+    '@typescript-eslint/no-explicit-any': 'off',
+    'no-unused-vars': 'off',
```
Comment (reno1979) [inline .eslintrc.js:22]: no-unused-vars is ignored? Interesting.

File: .storybook/preview.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import React from 'react';
 import { GlobalStyle, ThemeProvider } from '@amsterdam/asc-ui';
-import { ThemeProvider as MUIThemeProvider } from '@material-ui/core/styles';
-import muiTheme from '../src/theme/material-ui-theme';
+import { ThemeProvider as MUIThemeProvider } from '@mui/material';
+import { muiTheme } from '@amsterdam/bmi-component-library';
+// import muiTheme from '../src/theme/material-ui-theme';
```
Comment (reno1979) [inline .storybook/preview.js:5]: this one can be removed?

File: src/components/MetadataForm/MetadataFormStyles.ts [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+import { ComponentProps, FC } from 'react';
 import { Heading, themeSpacing } from '@amsterdam/asc-ui';
 
 export const MetadataFormStyle = styled.form`
 	display: flex;
 	flex-direction: column;
 `;
 
-export const StyledHeading = styled(Heading)<ComponentProps<typeof Heading>>`
+export const StyledHeading: FC<ComponentProps<typeof Heading>> = styled(Heading)`
```
Comment (reno1979) [inline src/components/MetadataForm/MetadataFormStyles.ts:10]: Should we type on React.FC ? or React.FunctionComponent

File: src/hooks/useCustomControl.tsx [hunk 4]
```diff
@@ -13,15 +13,15 @@ function useCustomControl({ data: value, errors, handleChange, path, schema }: C
 	}, [path]);
 
 	const onBlur = useCallback(
-		(e) => {
+		(e: any) => {
```
Comment (reno1979) [inline src/hooks/useCustomControl.tsx:16]: any ?

File: src/components/MetadataForm/MetadataFormStyles.ts (hunk 3 above)
Comment (fortunato) [reply src/components/MetadataForm/MetadataFormStyles.ts:10]: I wasn't able to get rid of the TS error after upgrading to the latest styled-components when overriding an already styled component in another way :man_shrugging:

**Review by reno1979 — APPROVED** [review]
geen dealbreakers, flinke klus geweest

File: .eslintrc.js (hunk 1 above)
Comment (fortunato) [reply .eslintrc.js:22]: Agreed. Weird. Not my doing I think

File: src/hooks/useCustomControl.tsx (hunk 4 above)
Comment (fortunato) [reply src/hooks/useCustomControl.tsx:16]: I had a go at guessing a type but it's variant. I decided to unblock myself and accept already existent technical debt.

**Pull request merged**